test/cli-spec.js

```diff
@@ -190,21 +192,53 @@ describe('command line interface', function () {
     );
   });
   it('should unpack static framework with all underlying symlinks unpacked', async () => {
-    const { tmpPath } = createSymlinkApp('app');
+    const { testPath } = await createSymlinkApp('app');
     await execAsar(
-      `p ${tmpPath} tmp/packthis-with-symlink.asar --unpack *.txt --unpack-dir var --exclude-hidden`,
+      `p ${testPath} tmp/packthis-with-symlink.asar --unpack *.txt --unpack-dir var --exclude-hidden`,
     );
 
-    assert.ok(fs.existsSync('tmp/packthis-with-symlink.asar.unpacked/private/var/file.txt'));
-    assert.ok(fs.existsSync('tmp/packthis-with-symlink.asar.unpacked/private/var/app/file.txt'));
-    assert.strictEqual(
-      fs.readlinkSync('tmp/packthis-with-symlink.asar.unpacked/private/var/app/file.txt'),
-      path.join('..', 'file.txt'),
+    await verifySmartUnpack('tmp/packthis-with-symlink.asar');
+  });
+  it('should respect ordering file (format: "${filepath}")', async () => {
+    const { testPath, filesOrdering } = await createSymlinkApp('app');
+
+    const orderingPath = path.join(testPath, '../ordered-app-ordering1.txt');
+    const data = filesOrdering.reduce((prev, curr) => {
+      return `${prev}${curr}\n`;
+    }, '');
+    await fs.writeFile(orderingPath, data);
+
+    await execAsar(
+      `p ${testPath} tmp/packthis-with-symlink.asar --ordering=${orderingPath} --exclude-hidden`,
     );
-    assert.strictEqual(
-      fs.readlinkSync('tmp/packthis-with-symlink.asar.unpacked/var'),
-      path.join('private', 'var'),
+    await verifySmartUnpack('tmp/packthis-with-symlink.asar');
+  });
+  it('should respect ordering file (format: ": ${filepath}")', async () => {
+    const { testPath, filesOrdering } = await createSymlinkApp('app');
+
+    const orderingPath = path.join(testPath, '../ordered-app-ordering2.txt');
+    const data = filesOrdering.reduce((prev, curr) => {
+      return `${prev}: ${curr}\n`;
+    }, '');
+    await fs.writeFile(orderingPath, data);
+
+    await execAsar(
+      `p ${testPath} tmp/packthis-with-symlink.asar --ordering=${orderingPath} --exclude-hidden`,
+    );
+    await verifySmartUnpack('tmp/packthis-with-symlink.asar');
+  });
+  it('should respect ordering file (format: "${random number} : ${filepath}")', async () => {
+    const { testPath, filesOrdering } = await createSymlinkApp('app');
+
+    const orderingPath = path.join(testPath, '../ordered-app-ordering3.txt');
+    const data = filesOrdering.reduce((prev, curr) => {
+      return `${prev}${Math.floor(Math.random() * 1000)} :  ${curr} \n`;
```

Comment:
question: what happens if the random number has a collision?

Reply:
Nothing, current logic ignores anything before the `:`
https://github.com/electron/asar/blob/9b7ccfec05292d47ea82cb44cd45339bc058febc/src/asar.ts#L93-L95